### PR TITLE
Integrate AdaNDV Estimator into NDVEstimator

### PR DESCRIPTION
**Pull request by Sherhom**
## Pull Request Summary

[Algo][Stats] Add AdaNDV integration into NDVEstimator

## Related Issues

Resolves: *(none yet)*

## Detailed Description

This PR adds the initial scaffolding for integrating the **AdaNDV** learned estimator into the NDVEstimator module of VIDEX.

### What has been done:
- Added `AdaNDVConfig` class for configuration management.
- Added `AdaNDVPredictor` class to load and run a pretrained AdaNDV model.
- Integrated `AdaNDV` as an estimation method in `NDVEstimator.estimator(...)`.
- Designed interface to compute prediction based on traditional estimator outputs.

### Limitations:
- Base estimators required by AdaNDV (e.g., EB, Chao, Jackknife, etc.) are not yet implemented.
- `sample_rate` is currently hardcoded; not inferred from actual data.
- No functional testing or verification yet.
- Not validated with actual NDV estimation queries or pipelines.

### Next Steps:
- Implement or adapt the 14 required base estimators (ref: `pydistinct`, original AdaNDV paper).
- Derive `sample_rate` dynamically from sample size.
- Verify AdaNDV accuracy on benchmark data.
- Add unit/integration tests.

---

<details>

<h3>Submission Checklist</h3>
<ul>
    <li>[x] PR title includes appropriate prefix(es)</li>
    <li>[x] Changes are clearly explained in the PR description</li>
    <li>[ ] New and existing tests pass successfully</li>
    <li>[ ] Code adheres to project style and best practices</li>
    <li>[ ] Documentation updated to reflect changes (if applicable)</li>
    <li>[ ] Changes have been tested on both Plugin-Mode and Standalone-Mode (if applicable)</li>
    <li>[ ] Statistical accuracy has been verified (for algorithm or optimizer changes)</li>
    <li>[ ] No regression in query plan accuracy compared to InnoDB (if applicable)</li>
    <li>[ ] Performance benchmarks conducted (for performance-sensitive changes)</li>
</ul>

</details>

### CODE REVIEW

File: src/sub_platforms/sql_opt/histogram/adandv_model_infer.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        self.model.load_state_dict(torch.load(config.model_path, map_location=device))
+        self.model.eval()
+
+    def predict(self, profile, estimate_list):
+        pad_len = self.config.model_input_len - 3  # 实验默认=97
+        profile = profile[:pad_len] + [0] * (pad_len - len(profile))
+
+        logn = np.log(np.dot(np.arange(len(profile)), profile))
+        logd = np.log(np.sum(profile))
+        logN = np.log(np.dot(np.arange(len(profile)), profile) / self.config.sample_rate)
```
Comment (xianghongxu) [inline src/sub_platforms/sql_opt/histogram/adandv_model_infer.py]: Use the table rows as the input, and logN can be directly obtained. Because the data type of sample rows n is an integer (**rounded**), using the sample rate to compute N may not be correct.

File: src/sub_platforms/sql_opt/histogram/adandv_model_infer.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import numpy as np
+
+
+class AdaNDVConfig:
+    def __init__(self,
+                 model_path: str,
+                 model_input_len: int = 100,
+                 estimator_num: int = 14,
+                 k: int = 2,
+                 sample_rate: float = 0.01):
```
Comment (xianghongxu) [inline src/sub_platforms/sql_opt/histogram/adandv_model_infer.py]: It is recommended to use table rows as the input
